### ISSUE TITLE
If stormcrawler above 2.5 uses Jdk 11 why the archetypes pom are not updated to 11

### ISSUE THREAD

**Issue by msghasan** [DigitalPebble/storm-crawler#1022]
The pom files in the archetype should be update to java 11 version rather than jdk 8
https://github.com/DigitalPebble/storm-crawler/blob/master/external/elasticsearch/archetype/src/main/resources/archetype-resources/pom.xml

**Comment by jnioche** [comment]
good catch, thanks @msghasan. Any chance you could submit a PR to fix it for the ES archetype but also the Opensearch and the core one?

**Comment by msghasan** [comment]
Sure will do raise a pr request

Maimur Hasan
Ph:-9962857984

**Comment by msghasan** [comment]
@jnioche  Hi I forked the repository to update the archtypes but there is a code the the core pom which is checking the existing code in what it is built i.e. java class version 52 and due to that check while I try to commit it is throwing me an error stating class version not supported... have you guys compiled the full project with java 11 once and deployed back to the repository.
[core pom file](https://github.com/DigitalPebble/storm-crawler/blob/master/pom.xml)
line no 120

**Comment by rzo1** [comment]
@msghasan I just changed the three POM files, updated `1.8` to `11` and conducted a fuil build (`clean install`) with Java 11 locally -> worked fine. Maybe you need to conduct a full build after your changes.

I noticed, that the build doesn't work with Java 17, yet (due to some module issues with incompatible maven plugins). @jnioche If we want to be able to build with Java 17, I can have a look via a separate issue. In addition, we have a dependency with a `<repository>` using `http` only, which will fail with newer Maven version (http-blocker). Might be also a thing to investigate :)

```bash
Downloading from maven-default-http-blocker: http://0.0.0.0/net/minidev/json-smart/maven-metadata.xml
[WARNING] Could not transfer metadata net.minidev:json-smart/maven-metadata.xml from/to maven-default-http-blocker (http://0.0.0.0/): transfer failed for http://0.0.0.0/net/minidev/json-smart/maven-metadata.xml
[WARNING] The POM for io.confluent:kafka-avro-serializer:jar:1.0 is missing, no dependency information available
```

**Comment by msghasan** [comment]
@rzo1 For me also the built went perfectly with java 11 when running build via terminal , but while committing from github desktop it throwing an error stating that the previous class version that the project was compile was with version 52 and the current version is 55

![image](https://user-images.githubusercontent.com/32441552/210213570-ccd9282b-b9e5-4711-ba1d-aa772db57571.png)

**Comment by msghasan** [comment]
I have java_home set is openjdk 11 and maven 3.8.5

**Comment by msghasan** [comment]
shall I reinstall github desktop and try again

**Comment by msghasan** [comment]
No, it doesn't.

Maimur Hasan
Ph:-9962857984

**Comment by rzo1** [comment]
Interesting. I am able to commit that change to my fork: https://github.com/rzo1/storm-crawler/commit/ed50c7a9e486c83a6286e7f4d70eba174c77d1f8 - no idea why it fails on your machine, though.

**Comment by msghasan** [comment]
Are you using eclipse or any other ide... I am using eclipse shall i try intellij

**Comment by rzo1** [comment]
I am using IntelliJ IDEA Community Edition on a Ubuntu 20.04 LTS machine. Don't think IDE makes a difference as the build seems to fail because of an too old runtime set for the Maven build.

**Comment by jnioche** [comment]
> I noticed, that the build doesn't work with Java 17, yet (due to some module issues with incompatible maven plugins). @jnioche If we want to be able to build with Java 17, I can have a look via a separate issue. 

Yes please. Not sure we want to move to 17 in the short term but anticipating the issues wouldn't hurt.

> In addition, we have a dependency with a <repository> using http only, which will fail with newer Maven version (http-blocker). Might be also a thing to investigate :)

Good idea. I suspect this has to do with the WARC module and probably a dependency that we can safely blacklist.

**Comment by jnioche** [comment]
Implemented in #1029
